internal/provwasm/stargate_whitelist.go

```diff
@@ -0,0 +1,163 @@
+
+	// bank
+	setWhitelistedQuery("/cosmos.bank.v1beta1.Query/Balance", &banktypes.QueryBalanceResponse{})
+	setWhitelistedQuery("/cosmos.bank.v1beta1.Query/DenomMetadata", &banktypes.QueryDenomsMetadataResponse{})
+	setWhitelistedQuery("/cosmos.bank.v1beta1.Query/Params", &banktypes.QueryParamsResponse{})
+	setWhitelistedQuery("/cosmos.bank.v1beta1.Query/SupplyOf", &banktypes.QuerySupplyOfResponse{})
+
+	// distribution
+	setWhitelistedQuery("/cosmos.distribution.v1beta1.Query/Params", &distributiontypes.QueryParamsResponse{})
+	setWhitelistedQuery("/cosmos.distribution.v1beta1.Query/DelegatorWithdrawAddress", &distributiontypes.QueryDelegatorWithdrawAddressResponse{})
+	setWhitelistedQuery("/cosmos.distribution.v1beta1.Query/ValidatorCommission", &distributiontypes.QueryValidatorCommissionResponse{})
+
+	// gov
+	setWhitelistedQuery("/cosmos.gov.v1beta1.Query/Deposit", &govtypes.QueryDepositResponse{})
+	setWhitelistedQuery("/cosmos.gov.v1beta1.Query/Params", &govtypes.QueryParamsResponse{})
+	setWhitelistedQuery("/cosmos.gov.v1beta1.Query/Vote", &govtypes.QueryVoteResponse{})
+
+	// slashing
+	setWhitelistedQuery("/cosmos.slashing.v1beta1.Query/Params", &slashingtypes.QueryParamsResponse{})
+	setWhitelistedQuery("/cosmos.slashing.v1beta1.Query/SigningInfo", &slashingtypes.QuerySigningInfoResponse{})
+
+	// staking
+	setWhitelistedQuery("/cosmos.staking.v1beta1.Query/Delegation", &stakingtypes.QueryDelegationResponse{})
+	setWhitelistedQuery("/cosmos.staking.v1beta1.Query/Params", &stakingtypes.QueryParamsResponse{})
+	setWhitelistedQuery("/cosmos.staking.v1beta1.Query/Validator", &stakingtypes.QueryValidatorResponse{})
+
+	// ==========================================================
+	// provenance queries
+	// ==========================================================
+
+	// attribute
+	setWhitelistedQuery("/provenance.attribute.v1.QueryParamsRequest", &attributetypes.QueryParamsResponse{})
+	setWhitelistedQuery("/provenance.attribute.v1.QueryAttributeRequest", &attributetypes.QueryAttributeResponse{})
+	setWhitelistedQuery("/provenance.attribute.v1.QueryAttributesRequest", &attributetypes.QueryAttributesResponse{})
+	setWhitelistedQuery("/provenance.attribute.v1.QueryScanRequest", &attributetypes.QueryScanResponse{})
+	setWhitelistedQuery("/provenance.attribute.v1.QueryAttributeAccountsRequest", &attributetypes.QueryAttributeAccountsResponse{})
+
+	// marker
+	setWhitelistedQuery("/provenance.marker.v1.QueryParamsRequest", &markertypes.QueryParamsResponse{})
+	setWhitelistedQuery("/provenance.marker.v1.QueryAllMarkersRequest", &markertypes.QueryAllMarkersResponse{})
+	setWhitelistedQuery("/provenance.marker.v1.QueryMarkerRequest", &markertypes.QueryMarkerResponse{})
+	setWhitelistedQuery("/provenance.marker.v1.QueryHoldingRequest", &markertypes.QueryHoldingResponse{})
+	setWhitelistedQuery("/provenance.marker.v1.QuerySupplyRequest", &markertypes.QuerySupplyResponse{})
```

Comment:
I think the paths you have might be incorrect. If you compare them to the cosmos requests, then you will notice a difference at the end of the message. The end should be /message.

For example, here is a comparison:
/cosmos.bank.v1beta1.Query/Balance
/provenance.marker.v1.QuerySupplyRequest

When I tested sending this message from a smart contract it had the following path.
"/provenance.marker.v1.Query/Supply"

Reply:
interesting... ok i'll fix that